### DISCUSSION TITLE
Ruby/Golang bindings

### DISCUSSION THREAD

**Discussion by arjunmenon** [slint-ui/slint#5049] [Q&A]
Hey, is it possible to use Slint with Ruby or Golang? Any plans for it or any guide to generate bindings for them?

**Comment by hunger** [comment]
We officially support rust, c++, js and python at this time, but I have seen people blog about other languages like c# before though. I am not aware of any go or ruby bindings at this time though.

I think we would like to look into go at some point (no concrete plans though!), but I have not heard ruby being brought up yet.

We are absolutely open to add more languages of course.

You have two options to integrate a language:

* as a compilation target
* using the interpreter

A dynamic language like ruby will probably be served better with the second approach while the first one will probably fit Go better...

Examples for the compiler backend are in `internal/compiler/generator` if you are looking for inspiration, while the js and python code is in `api/python` and in `api/node`.

The mattermost server in chat.slint.dev is a nice way to ask questions (related to this or other topics). We can chat there and even spin up a voice or video chat from there :-)

**Comment by ddkwork** [comment]
https://github.com/can1357/gengo

**Reply by hunger** [reply]
A C library wrapper for go can help get to get this effort started, but additional work will be required to make slint feel "native" to go developers.

**Reply by ddkwork** [reply]
For go, Windows platform cgo compilation speed is too slow, so I recommend using this library to bind c header files, at the same time, need to compile slint as a shared dynamic library for each platform, so that cross-compilation is easy and compilation speed is close to native. The library I recommend is suitable for calling dynamic libraries on all platforms, and I have verified its feasibility in another repository.

https://github.com/ddkwork/gui/blob/main/sdk%2Fsdk.go

https://github.com/ddkwork/glfw/tree/main

**Comment by vovavili** [comment]
Having Slint work with a fast and well-established AOT compiled language with a garbage collector like Go would be beyond fantastic. I feel like C++, Rust, Go, JavaScript and Python cover the vast majority of relevant developments scenarios for graphical software in 2025, Go basically fills the spot for "I need my software to be fast or otherwise compiled, but I don't want to deal with manual memory management".

**Comment by jclab-joseph** [comment]
Vote! +1

**Comment by jclab-joseph** [comment]
It is a big change, so it might be difficult to merge, but I am working on it personally.
https://github.com/slint-ui/slint/pull/11308
